
Property changes on: .
___________________________________________________________________
Modified: svn:ignore
   - .classpath
.project
experiments
hudson.iml
hudson.ipr
hudson.iws
ref
svn-commit*.tmp
work
target

   + .classpath
.project
experiments
hudson.iml
hudson.ipr
hudson.iws
ref
svn-commit*.tmp
work
target
.metadata


Index: main/core/src/main/java/hudson/matrix/LinkedLogRotator.java
===================================================================
--- main/core/src/main/java/hudson/matrix/LinkedLogRotator.java	(revision 32527)
+++ main/core/src/main/java/hudson/matrix/LinkedLogRotator.java	(working copy)
@@ -28,6 +28,11 @@
 
 import java.io.IOException;
 
+
+import static java.util.logging.Level.FINER;
+
+import java.util.logging.Logger;
+
 /**
  * {@link LogRotator} for {@link MatrixConfiguration},
  * which discards the builds if and only if it's discarded
@@ -40,21 +45,37 @@
  * @author Kohsuke Kawaguchi
  */
 final class LinkedLogRotator extends LogRotator {
+    /**
+     * @deprecated
+     *      Use {@link #LinkedLogRotator(int, int)}
+     */
     LinkedLogRotator() {
         super(-1,-1);
     }
+    LinkedLogRotator(int artifactDaysToKeep, int artifactNumToKeep) {
+        super(-1,-1, artifactDaysToKeep, artifactNumToKeep);
+    }
 
     @Override
     public void perform(Job _job) throws IOException, InterruptedException {
         // copy it to the array because we'll be deleting builds as we go.
         MatrixConfiguration job = (MatrixConfiguration) _job;
+        /* let superclass handle clearing artifacts if defined. */
+        super.perform(_job);
 
         for( MatrixRun r : job.getBuilds().toArray(new MatrixRun[0]) ) {
             if(job.getParent().getBuildByNumber(r.getNumber())==null)
+            {
+                LOGGER.log(FINER,r.getFullDisplayName()+" is GC-ed because it's the parent build is non existant.");
                 r.delete();
+            }
         }
 
         if(!job.isActiveConfiguration() && job.getLastBuild()==null)
+        {
+            LOGGER.log(FINEER,job.getFullDisplayName()+" is deleted as not active and no builds left.");
             job.delete();
+        }
+            
     }
 }
Index: main/core/src/main/java/hudson/matrix/MatrixConfiguration.java
===================================================================
--- main/core/src/main/java/hudson/matrix/MatrixConfiguration.java	(revision 32527)
+++ main/core/src/main/java/hudson/matrix/MatrixConfiguration.java	(working copy)
@@ -240,7 +240,9 @@
 
     @Override
     public LogRotator getLogRotator() {
-        return new LinkedLogRotator();
+        int archivedaystokeep = getParent().getLogRotator().getArtifactDaysToKeep();
+        int archivenumbertokeep = getParent().getLogRotator().getArtifactNumToKeep();
+        return new LinkedLogRotator(archivedaystokeep, archivenumbertokeep);
     }
 
     @Override
Index: main/war/nbactions.xml
===================================================================
--- main/war/nbactions.xml	(revision 32527)
+++ main/war/nbactions.xml	(working copy)
@@ -21,12 +21,9 @@
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
--->
-
-<!--
+--><!--
 Ensures that NetBeans knows which goals to run when Run Project is selected.
--->
-<actions>
+--><actions>
     <action>
         <actionName>run</actionName>
         <goals>
@@ -42,4 +39,13 @@
             <jpda.listen>maven</jpda.listen>
         </properties>
     </action>
-</actions>
\ No newline at end of file
+    <action>
+            <actionName>build</actionName>
+            <goals>
+                <goal>install</goal>
+            </goals>
+            <properties>
+                <maven.test.skip>true</maven.test.skip>
+            </properties>
+        </action>
+</actions>
